### PR TITLE
CLOSED - Utils - Tweak - In cases of a multi-domain or a mirror domain, client-side assets will not be loaded due to CORS error - ED-1931

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: tests/phpunit/elementor/includes/test-utils.php [hunk 1]
```diff
@@ -180,4 +180,141 @@ public function test_is_empty() {
...
+		$this->get_current_domain_src__sub_domain();
+
+		// Scenario 2 - Subdomain with a root folder ( www.x.com/site === de.x.com/site ).
+		$this->get_current_domain_src__sub_domain_root_folder();
+
+		// Scenario 3 - Root folder + mirror domain points to root folder ( www.x.com/site === www.y.com ).
+		$this->get_current_domain_src__mirror_domain();
+	}
+
+	public function get_current_domain_src__set_site_url( $url = '' ) {
```
Comment (kobizz) [inline tests/phpunit/elementor/includes/test-utils.php:251]: A non-test method should be private and at the end of the class

File: tests/phpunit/elementor/includes/test-utils.php [hunk 2]
```diff
@@ -180,4 +180,141 @@ public function test_is_empty() {
...
+		return $url . '/my-website-folder';
+	}
+
+	public function get_current_domain_src__sub_domain() {
```
Comment (kobizz) [inline tests/phpunit/elementor/includes/test-utils.php:255]:
```suggestion
	public function test_get_current_domain_src__sub_domain() {
```

File: tests/phpunit/elementor/includes/test-utils.php [hunk 3]
```diff
@@ -180,4 +180,141 @@ public function test_is_empty() {
...
+			'http://de.example.org/wp-content/plugins/elementor/assets/css/common.css?ver=3.1.0',
+			Utils::get_current_domain_src("http://$initial_server_name/wp-content/plugins/elementor/assets/css/common.css?ver=3.1.0" ),
+			'sub_domain - Failed to replace current domain with subdomain.'
+		);
+
+		// Restoring SERVER_NAME value.
+		$_SERVER['SERVER_NAME'] = $initial_server_name;
+	}
+
+	public function get_current_domain_src__sub_domain_root_folder() {
```
Comment (kobizz) [inline tests/phpunit/elementor/includes/test-utils.php:271]:
```suggestion
	public function test_get_current_domain_src__sub_domain_root_folder() {
```

File: tests/phpunit/elementor/includes/test-utils.php [hunk 4]
```diff
@@ -180,4 +180,141 @@ public function test_is_empty() {
...
+		remove_filter( 'site_url', 'test_get_current_domain_src__set_root_folder', 10, 4 );
+
+		// Restoring root folder.
+		$_SERVER['PHP_SELF'] = $initial_php_self;
+
+		// Restoring SERVER_NAME value.
+		$_SERVER['SERVER_NAME'] = $initial_server_name;
+	}
+
+	public function get_current_domain_src__mirror_domain() {
```
Comment (kobizz) [inline tests/phpunit/elementor/includes/test-utils.php:300]:
```suggestion
	public function test_get_current_domain_src__mirror_domain() {
```

File: tests/phpunit/elementor/includes/test-utils.php [hunk 5]
```diff
@@ -180,4 +180,141 @@ public function test_is_empty() {
...
+			'http://www.mirror-domain.com/wp-content/plugins/elementor/assets/lib/swiper/swiper.js?ver=5.3.6',
+			Utils::get_current_domain_src("http://$initial_server_name/my-website-folder/wp-content/plugins/elementor/assets/lib/swiper/swiper.js?ver=5.3.6" ),
+			'mirror_domain - Failed to replace current domain with mirror domain.'
+		);
+
+		remove_filter( 'site_url', 'test_get_current_domain_src__set_root_folder', 10, 4 );
+
+		// Restoring SERVER_NAME value.
+		$_SERVER['SERVER_NAME'] = $initial_server_name;
+	}
```
Comment (kobizz) [inline tests/phpunit/elementor/includes/test-utils.php:319]: Please add tests for scenarios like these: `http://localhost/`, `http://domain.com/`

**Review by kobizz — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/includes/test-utils.php (hunk 5 above)
Comment (rotemee) [reply tests/phpunit/elementor/includes/test-utils.php:319]: As agreed, it's already covered in our current tests.

File: tests/phpunit/elementor/includes/test-utils.php (hunk 1 above)
Comment (rotemee) [reply tests/phpunit/elementor/includes/test-utils.php:251]: It must be public to be used in a filter.

File: tests/phpunit/elementor/includes/test-utils.php (hunk 2 above)
Comment (rotemee) [reply tests/phpunit/elementor/includes/test-utils.php]: Fixed.

File: tests/phpunit/elementor/includes/test-utils.php (hunk 3 above)
Comment (rotemee) [reply tests/phpunit/elementor/includes/test-utils.php]: Fixed.

File: tests/phpunit/elementor/includes/test-utils.php (hunk 4 above)
Comment (rotemee) [reply tests/phpunit/elementor/includes/test-utils.php]: Fixed.

**Pull request closed without merging**